### PR TITLE
Add shadows for the Virtual Keyboard buttons

### PR DESCRIPTION
**Pull request by Districh-ru**
This PR adds shadows for the Virtual keyboard buttons

virtKB

### CODE REVIEW

File: src/fheroes2/gui/ui_keyboard.cpp [hunk 1]
```diff
@@ -591,7 +602,13 @@ namespace
 
         fheroes2::ButtonSprite okayButton( windowRoi.x + ( windowRoi.width - okayButtonReleasedImage.width() ) / 2, windowRoi.y + windowRoi.height - 35,
```
Comment (ihhub) [inline src/fheroes2/gui/ui_keyboard.cpp:603]: I suggest to move `windowRoi.x + ( windowRoi.width - okayButtonReleasedImage.width() ) / 2, windowRoi.y + windowRoi.height - 35` to a separate Point variable as we use it in 2 places.

File: src/fheroes2/gui/ui_keyboard.cpp [hunk 2]
```diff
@@ -477,8 +479,9 @@ namespace
 
         fheroes2::Rect roi{ offset.x + offsets.front(), offset.y, 1, 1 };
 
-        int32_t yOffset = offset.y;
-        for ( size_t i = 0; i < buttonLayout.size(); ++i ) {
+        const size_t buttonRows = buttonLayout.size();
+
+        for ( size_t i = 0; i < buttonRows; ++i ) {
```
Comment (ihhub) [inline src/fheroes2/gui/ui_keyboard.cpp:484]: I am not sure that this change will make the code faster as `size()` method is inlined and marked as const expression.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I left 2 comments in this pull request. Would you mind to check them if you have spare time?

File: src/fheroes2/gui/ui_keyboard.cpp (hunk 2 above)
Comment (Districh-ru) [reply src/fheroes2/gui/ui_keyboard.cpp:484]: Hi, @ihhub, `buttonRows` is also used 2 times in the 497 line to I cached it as it it used here in the loop and in the line 497.

File: src/fheroes2/gui/ui_keyboard.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/gui/ui_keyboard.cpp]: Sure)

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]
Well done!

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for these changes!

**Pull request merged**